src/ert/analysis/_es_update.py

```diff
@@ -474,6 +478,14 @@ def adaptive_localization_progress_callback(
     num_obs = len(observation_values)
 
     smoother_snapshot.update_step_snapshots = update_snapshot
+    # Used as labels for observations in cross-correlation matrix.
+    # Say we have two observation groups "FOPR" and "WOPR" where "FOPR" has
+    # 2 responses and "WOPR" has 3.
+    # In this case we create a list [FOPR_0, FOPR_1, WOPR_0, WOPR_1, WOPR_2]
```

Comment:
Would it be possible to use the index for the observations here instead? I guess it is ok for summary observations, though perhaps a bit confusing as this number will not align with `report_step`, but for `GEN_OBS` which already have an int index it would probably not align with that, and could be confusing.

Reply:
Not sure. Do I remember correctly that the index is a time-stamp or date for summary observations? We could perhaps still use it but perhaps a bit messy.

Reply:
Yes, was thinking a bit more about this, and should we consider doing it like we do for the snapshot in the gui showing the scaling factors when doing auto scaling? Those events are automatically written to csv by the client, so will be available to the user.

Something like: https://github.com/equinor/ert/blob/71275280104b46942e41a907a76342e262928aeb/src/ert/analysis/_es_update.py#L239, which alread uses the indices: https://github.com/equinor/ert/blob/71275280104b46942e41a907a76342e262928aeb/src/ert/analysis/_es_update.py#L252

Reply:
As far as I know, these are the two types of indexes we have:

<img width="543" alt="Screenshot 2025-01-08 at 09 02 25" src="https://github.com/user-attachments/assets/9e3302c3-c66f-4652-8b4f-ef94b76634c4" />

<img width="551" alt="Screenshot 2025-01-08 at 09 03 47" src="https://github.com/user-attachments/assets/12e8a562-f697-49f2-8b52-7b25c45cbf39" />

Are you suggesting we use the second part of the string as an index?

Reply:
> Are you suggesting we use the second part of the string as an index?

Yes, or perhaps send the whole thing as an event instead of saving it in storage. Then everyone gets the benefit right away.

Reply:
Not sure I follow the event thing. Would we then still be able to analyse the data from a Jupyter Notebook?